### ISSUE TITLE
Flag effect remains active outside of current group

### ISSUE THREAD

**Issue by okdana** [rust-lang/regex#640]
The [`regex` documentation](https://docs.rs/regex/1.3.1/regex/#grouping-and-flags) suggests that flags (`(?foo)`) only apply within the 'current group', which would match the behaviour of Perl and PCRE, but that doesn't seem to be the case:

```
# With PCRE
% printf '%s\n' foo Foo bar Bar | \rg -sP '((?i)foo)|Bar'
foo
Foo
Bar

# With Rust regex
% printf '%s\n' foo Foo bar Bar | \rg -s '((?i)foo)|Bar'
foo
Foo
bar  # ???
Bar
```

(My test case uses ripgrep 11.0.2 from Homebrew, but i've confirmed that it works the same way if i compile a simple test app with `regex` 1.3.3.)

Am i misunderstanding something, or is this wrong?

**Comment by BurntSushi** [comment]
Yeesh. What a bug. This is definitely wrong. Interestingly, the code was actually going out of its way not to reset flags in this case. I'm not sure why. I don't think I ever intended for this behavior, and no tests were checking for it. And as you say, the docs seem to pretty strongly contradict the current behavior.

I have a fix up. Thanks for reporting this!

**Comment by BurntSushi** [comment]
OK, this should be fixed in `regex 1.3.4` and `regex-syntax 0.6.14` on crates.io. I've also updated ripgrep master to include these dependency updates.

**Comment by okdana** [comment]
Thanks!
